android/src/main/java/fr/greweb/reactnativeviewshot/ViewShot.java

```diff
@@ -200,32 +172,13 @@ public ViewShot(
         this.currentActivity = currentActivity;
         this.handleGLSurfaceView = handleGLSurfaceView;
         this.promise = promise;
-
-        reactContext.addLifecycleEventListener(this);
-
-        // bg hanadler for non UI heavy work
-        mBgThread = new HandlerThread("RNViewShot-Handler-Thread");
-        mBgThread.start();
-        mBgHandler = new Handler(mBgThread.getLooper());
     }
     //endregion
 
-    private void cleanup() {
-        if (mBgThread != null) {
-            if (Build.VERSION.SDK_INT < 18) {
-                mBgThread.quit();
-            } else {
-                mBgThread.quitSafely();
-            }
-
-            mBgThread = null;
-        }
-    }
-
     //region Overrides
     @Override
     public void execute(final NativeViewHierarchyManager nativeViewHierarchyManager) {
-        mBgHandler.post(new Runnable() {
+        new Thread("RNViewShot-Capture-Thread") {
```

Comment:
are you sure this is safe to do this in Android? I fear about thread leaks & usually isn't it why we need a mem pool?

Reply:
It should be safe, but it's not really the best practice. Reimplemented with a pool